website/cue/reference/components/sinks/base/aws_cloudwatch_logs.cue

```diff
@@ -0,0 +1,448 @@
+				required:    false
+				type: uint: {}
+			}
+			profile: {
+				description: "The credentials profile to use."
+				required:    false
+				type: string: syntax: "literal"
+			}
+			region: {
+				description: """
+					The AWS region to send STS requests to.
+
+					If not set, this will default to the configured region
+					for the service itself.
+					"""
+				required: false
+				type: string: syntax: "literal"
+			}
+			secret_access_key: {
+				description: "The AWS secret access key."
+				required:    true
+				type: string: syntax: "literal"
+			}
+		}
+	}
+	batch: {
+		description: "Event batching behavior."
+		required:    false
+		type: object: options: {
+			max_bytes: {
+				description: """
+					The maximum size of a batch that will be processed by a sink.
+
+					This is based on the uncompressed size of the batched events, before they are
+					serialized / compressed.
+					"""
+				required: false
+				type: uint: {}
+			}
+			max_events: {
+				description: "The maximum size of a batch, in events, before it is flushed."
+				required:    false
+				type: uint: {}
+			}
+			timeout_secs: {
+				description: "The maximum age of a batch, in seconds, before it is flushed."
+				required:    false
+				type: float: {}
+			}
+		}
+	}
+	compression: {
+		description: "Compression configuration."
+		required:    false
+		type: {
+			object: options: {
+				algorithm: {
+					required: false
+					type: string: {
+						const:   "zlib"
+						default: "none"
+					}
+				}
+				level: {
+					description: "Compression level."
+					required:    false
+					type: {
+						number: enum: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+						string: enum: ["none", "fast", "best", "default"]
+					}
+				}
+			}
+			string: enum: ["none", "gzip", "zlib"]
+		}
+	}
+	create_missing_group: {
+		description: """
+			Dynamically create a [log group][log_group] if it does not already exist.
+
+			This will ignore `create_missing_stream` directly after creating the group and will create
+			the first stream.
+
+			[log_group]: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Working-with-log-groups-and-streams.html
+			"""
+		required: false
+		type: bool: {}
+	}
+	create_missing_stream: {
+		description: """
+			Dynamically create a [log stream][log_stream] if it does not already exist.
+
+			[log_stream]: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/Working-with-log-groups-and-streams.html
+			"""
+		required: false
+		type: bool: {}
+	}
+	encoding: {
+		description: "Encoding configuration."
+		required:    true
+		type: object: options: {
+			avro: {
+				description:   "Apache Avro serializer options."
+				relevant_when: "codec = \"avro\""
+				required:      true
+				type: object: options: schema: {
+					description: "The Avro schema."
+					required:    true
+					type: string: syntax: "literal"
+				}
+			}
+			codec: {
+				required: true
+				type: string: enum: {
```

Comment:
Were these all always "allowed" but not previously documented?

Reply:
Yep.

We selectively controlled which of them got emitted in Cue and thus added to the documentation, but since it the same ol' `Encoding` field in the component's configuration type... whatever it supported was possible to use.